packages/teams-js/src/internal/appHelpers.ts

```diff
@@ -70,8 +87,37 @@ export function notifyFailureHelper(apiVersiontag: string, appInitializationFail
   ]);
 }
 
-export function notifySuccessHelper(apiVersionTag: string): void {
-  sendMessageToParent(apiVersionTag, app.Messages.Success, [version]);
+function supportsNotifySuccessResponse(): boolean {
+  return ensureInitialized(runtime) && !!runtime.supports.app?.notifySuccessResponse;
+}
+
+export async function notifySuccessHelper(apiVersionTag: string): Promise<NotifySuccessResponse> {
+  // If the initialize already completed, dispatch notify success
+  if (GlobalVars.initializeCompleted) {
+    return callNotifySuccessInHost(apiVersionTag);
+  }
+
+  // If initialize is still waiting for response, dispatch the call after initialize
+  // finishes to have the full runtime object instantiated.
+  if (!GlobalVars.initializePromise) {
+    throw new Error(errorLibraryNotInitialized);
+  }
+  return GlobalVars.initializePromise.then(() => callNotifySuccessInHost(apiVersionTag));
+}
+
+export async function callNotifySuccessInHost(apiVersionTag: string): Promise<NotifySuccessResponse> {
+  if (!supportsNotifySuccessResponse()) {
+    callFunctionInHost(app.Messages.Success, [version], apiVersionTag);
```

Comment:
That makes sense, reading into the implementation I can see that function is waiting for the response. I think is a simple change reverting to the deprecated function. 

Still, there may be some needs in the future to add fire-and-forget type capabilities and it would be great to support that behavior in the "correct" way. Maybe we should add a backlog task for creating another function to do so? Or add a flag in the `callFunctionInHost` to be able to create fire-and-forget capabilities? I'll push the change, but I could include something like that if you want 😄

Reply:
I would like to wait to add a "correct" way to fire and forget from TeamsJS until we think of a legitimate reason to need one. So far, I haven't been able to think of a legitimate reason to ever call a function in the host that can't *at least* potentially receive errors from the host if something goes wrong (which is what `callFunctionInHost` is for).